source/scpp/src/scp/QuorumSetUtils.cpp

```diff
@@ -33,24 +34,50 @@ class QuorumSetSanityChecker
     bool mIsSane;
     size_t mCount{0};
 
-    bool checkSanity(SCPQuorumSet const& qSet, int depth);
+    bool checkSanity(SCPQuorumSet const& qSet, int depth, const char** reason);
 };
 
 QuorumSetSanityChecker::QuorumSetSanityChecker(SCPQuorumSet const& qSet,
-                                               bool extraChecks)
+                                               bool extraChecks,
+                                               const char** reason)
     : mExtraChecks{extraChecks}
 {
-    mIsSane = checkSanity(qSet, 0) && mCount >= 1 && mCount <= 1000;
+    mIsSane = checkSanity(qSet, 0, reason);
+
+    if (mCount < 1 || mCount > 1000)
+    {
+        if (reason != nullptr)
+        {
+            const char* msg = "Number of validator nodes exceeds the limit [1..1000]";
```

Comment:
Technically it's out of bounds, as it can be `0`.
You could also split it up into "Quorum set has no validator" and "Quorum set has over 1000 validators".